specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,267 @@
+                            showCustomizedDialog();
+                            CHECK_FAILURE(deferral->Complete());
+                        });
+                    return S_OK;
+                })
+                .Get(),
+            &m_saveAsRequestedToken);
+    }
+    m_silentSaveAs = !m_silentSaveAs;
+    MessageBox(
+        m_appWindow->GetMainWindow(),
+        (m_silentSaveAs ? L"Silent Save As Enabled" : L"Silent Save As Disabled"), L"Info",
+        MB_OK);
+    return true;
+}
+//! [ToggleSilent]
+
+//! [ProgrammaticSaveAs]
+// Call SaveContentAs method to trigger the programmatic save as.
+bool ScenarioSaveAs::ProgrammaticSaveAs()
+{
+    if (!m_webView2Staging20)
+        return false;
+    m_webView2Staging20->SaveContentAs(
+        Callback<ICoreWebView2StagingSaveContentAsCompletedHandler>(
+            [this](HRESULT errorCode, COREWEBVIEW2_SAVE_CONTENT_AS_RESULTS result) -> HRESULT
+            {
+                // Show SaveContentAs returned result, optional
+                // Skip message box when the result is COREWEBVIEW2_SAVE_AS_OPEN_SYSTEM_DIALOG (0),
+                // to prevent it deactivates save as modal dialog focus on the window
+                if (result > 0)
+                {
+                    MessageBox(
+                        m_appWindow->GetMainWindow(),
+                        (L"Save As " + saveAsResultString[result]).c_str(), L"Info", MB_OK);
+                }
+                return S_OK;
+            })
+            .Get());
+    return true;
+}
+//! [ProgrammaticSaveAs]
+```
+# API Details
+## Win32 C++
+```c++
+/// Specifies save as type selection options for `ICoreWebView2Staging20`,
+/// used in `SaveAsRequestedEventArgs`
+///
+/// When the source is a html page, allows to select `HTML_ONLY`,
+/// `SINGLE_FILE`, `COMPLETE`; when the source is a non-html,
+/// only allows to select `DEFAULT`; otherwise, will deny the download
+/// and return `COREWEBVIEW2_SAVE_AS_TYPE_NOT_SUPPORTED`.
+///
+/// The content type/format is a MIME type, indicated by the source
+/// server side and identified by the browser. It’s not related to the
+/// file’s type or extension. MIME type of `text/html`,
+/// `application/xhtml+xml` are considered as html page.
+[v1_enum] typedef enum COREWEBVIEW2_SAVE_AS_TYPE {
+  /// Default to save for a non-html content. If it is selected for a html
+  /// page, it’s same as HTML_ONLY option.
+  COREWEBVIEW2_SAVE_AS_TYPE_DEFAULT,
+  /// Save the page as html
+  COREWEBVIEW2_SAVE_AS_TYPE_HTML_ONLY,
+  /// Save the page as mhtml
+  COREWEBVIEW2_SAVE_AS_TYPE_SINGLE_FILE,
+  /// Save the page as html, plus, download the page related source files in
+  /// a folder
+  COREWEBVIEW2_SAVE_AS_TYPE_COMPLETE,
+} COREWEBVIEW2_SAVE_AS_TYPE;
+
+/// Status of a programmatic save as call, indicates the result
+/// for method `SaveContentAs`
+[v1_enum] typedef enum COREWEBVIEW2_SAVE_CONTENT_AS_RESULTS {
+  /// Programmatically open a system default save as dialog
```

Comment:
I can't tell from this text if this is an error, or something the end dev needs to do, or what?

I'm guessing its an error, like we can't show our dialog because the system default save as dialog is already shown?

```suggestion
  /// The system default save as dialog is already opened and an additional save as dialog cannot be opened.
```

Reply:
> developer should know this.

If the end developer chooses to show the default dialog or not and so knows they showed the dialog or not, then why do we have it? We can remove COREWEBVIEW2_SAVE_AS_OPEN_SYSTEM_DIALOG?